src/shapes/group.class.js

```diff
@@ -224,23 +220,53 @@
       object[directive]('deselected', this.__objectSelectionDisposer);
     },
 
+    /**
+     * Checks if object can enter group and logs relevant warnings
+     * @private
+     * @param {fabric.Object} object
+     * @returns
+     */
+    canEnter: function (object) {
+      if (object === this) {
+        /* _DEV_MODE_START_ */
+        console.warn('fabric.Group: trying to add group to itself, this call has no effect');
+        /* _DEV_MODE_END_ */
+        return false;
+      }
+      else if (object.group && object.group === this) {
+        /* _DEV_MODE_START_ */
+        console.warn('fabric.Group: duplicate objects are not supported inside group, this call has no effect');
+        /* _DEV_MODE_END_ */
+        return false;
+      }
```

Comment:
handled by
[3df9435](https://github.com/fabricjs/fabric.js/pull/7859/commits/3df9435e0861c0c16edf7d2c4b1f90c3eb451061)

Reply:
regarding [3df9435](https://github.com/fabricjs/fabric.js/pull/7859/commits/3df9435e0861c0c16edf7d2c4b1f90c3eb451061) 

First of all i m not sure if you thing the issue is cleare or not.
I see the issue for direct descendant, so for example.
c is a group included in B that is a group included in a, i try to check if `c.canLetEnter(a)`,  in that case running a.isDescendantOf(c) return false and we add it and we create a circular dependency in the._objects array.

Correct?

Can we check both?

Anyway performances cost outside the render loop do not matter in a meaningful way.

Reply:
> regarding [3df9435](https://github.com/fabricjs/fabric.js/pull/7859/commits/3df9435e0861c0c16edf7d2c4b1f90c3eb451061)
> 
> First of all i m not sure if you think the issue is cleared or not. I see the issue for non direct descendant, so for example. c is a group included in B that is a group included in a, i try to check if `a.canLetEnter(c)`, in that case running a.isDescendantOf(c) return false and we add it and we create a circular dependency in the._objects array.
> 
> Correct?

No because there are other forces at play. Part of the `enterGroup` logic removes c from b and only then truly accepts c into a.

The question is are there any more cases....
- if c is completely strange to a then it can enter
- if c is descendant of a before entering a then it will be removed from it's direct parent and will enter a
- if c is a direct descendant of a `canEnter` returns false.

Did I miss something? The rest of the possibilities aren't real. I mean these 3 conditions won't allow them to happen.


Regarding naming
`canLetEnter` is bad as well. What about `canEnterGroup` or `grantEntrance` or `isEnteringAllowed` or `assertEnterObject` (implying it throws)

Reply:
`allowEntrance`?

Reply:
This is related as well https://github.com/fabricjs/fabric.js/pull/7859#discussion_r846754553